CRM/Mailing/BAO/MailingAB.php

```diff
@@ -48,30 +48,27 @@ public static function create(&$params) {
    *
    * @param int $id
    *   Id of the mail to delete.
+   *
+   * @deprecated
    */
   public static function del($id) {
-    if (empty($id)) {
-      throw new CRM_Core_Exception(ts('No id passed to MailingAB del function'));
-    }
-    CRM_Core_Transaction::create()->run(function () use ($id) {
-      CRM_Utils_Hook::pre('delete', 'MailingAB', $id);
+    static::deleteRecord(['id' => $id]);
+    CRM_Core_Session::setStatus(ts('Selected mailing has been deleted.'), ts('Deleted'), 'success');
+  }
 
-      $dao = new CRM_Mailing_DAO_MailingAB();
-      $dao->id = $id;
-      if ($dao->find(TRUE)) {
-        $mailing_ids = [$dao->mailing_id_a, $dao->mailing_id_b, $dao->mailing_id_c];
-        $dao->delete();
-        foreach ($mailing_ids as $mailing_id) {
-          if ($mailing_id) {
-            CRM_Mailing_BAO_Mailing::del($mailing_id);
```

Comment:
Oh god I thought it couldn't get more insane but it can. This function was setting a status message and *also* calling `CRM_Mailing_BAO_Mailing::del` which sets another (redundant) status message! Horrors upon horrors! So calling this function is going to generate about 3 redundant status messages. Delete them! Quick, before they multiply!!!

Reply:
They're gone!

The old version has the `Mailing` deletes wrapped in a transaction.  I don't think the new version needs any extra transaction stuff does it?